docs/api/Errors.md

```diff
@@ -7,15 +7,19 @@ You can find all the error objects inside the `errors` key.
-| `RequestAbortedError`        |  `UND_ERR_ABORTED`                | the request has been aborted by the user       |
-| `ClientDestroyedError`       |  `UND_ERR_DESTROYED`              | trying to use a destroyed client.              |
-| `ClientClosedError`          |  `UND_ERR_CLOSED`                 | trying to use a closed client.                 |
-| `SocketError`                |  `UND_ERR_SOCKET`                 | there is an error with the socket.             |
-| `NotSupportedError`          |  `UND_ERR_NOT_SUPPORTED`          | encountered unsupported functionality.         |
-| `ContentLengthMismatchError` |  `UND_ERR_CONTENT_LENGTH_MISMATCH`| body does not match content-length header      |
-| `InformationalError`         |  `UND_ERR_INFO`                   | expected error with reason                     |
-| `TrailerMismatchError`       |  `UND_ERR_TRAILER_MISMATCH`       | trailers did not match specification           |
\ No newline at end of file
+| Error                        | Error Codes                                | Description                                           |
+| -----------------------------|--------------------------------------------|-------------------------------------------------------|
+| `InvalidArgumentError`       |  `UND_ERR_INVALID_ARG`                     | passed an invalid argument.                           |
+| `InvalidReturnValueError`    |  `UND_ERR_INVALID_RETURN_VALUE`            | returned an invalid value.                            |
+| `RequestAbortedError`        |  `UND_ERR_ABORTED`                         | the request has been aborted by the user              |
+| `ClientDestroyedError`       |  `UND_ERR_DESTROYED`                       | trying to use a destroyed client.                     |
+| `ClientClosedError`          |  `UND_ERR_CLOSED`                          | trying to use a closed client.                        |
+| `SocketError`                |  `UND_ERR_SOCKET`                          | there is an error with the socket.                    |
+| `NotSupportedError`          |  `UND_ERR_NOT_SUPPORTED`                   | encountered unsupported functionality.                |
+| `ResponseContentLength`      |  `UND_ERR_REQUEST_CONTENT_LENGTH_MISMATCH` | body does not match content-length header in request  |
+| `RequestContentLength`       |  `UND_ERR_RESPONSE_CONTENT_LENGTH_MISMATCH`| body does not match content-length header in response * |
+| `InformationalError`         |  `UND_ERR_INFO`                            | expected error with reason                            |
+| `TrailerMismatchError`       |  `UND_ERR_TRAILER_MISMATCH`                | trailers did not match specification                  |
+
+\* The `UND_ERR_RESPONSE_CONTENT_LENGTH_MISMATCH` is returned when the received body is less than the value in the received header `Content-Length`.  
```

Comment:
Please note that at https://github.com/nodejs/undici/blob/e2a9ddc07d9e0ff6be25d243523b53d1d71a3e9e/deps/llhttp/include/llhttp.h

If we want to allow body larger than content-length, we probably should expose this option